provider/k8s.go

```diff
@@ -1442,31 +1447,92 @@ func (iter *FileStoreFeatureIterator) Next() bool {
 	if nextVal == nil {
 		return false
 	}
-	formatDate := "2006-01-02 15:04:05 UTC" // hardcoded golang format date
-	timeString, ok := nextVal["ts"].(string)
-	if !ok {
-		iter.cur = ResourceRecord{Entity: fmt.Sprintf("%s", nextVal["entity"]), Value: nextVal["value"]}
-	} else {
-		timestamp, err1 := time.Parse(formatDate, timeString)
-		formatDateWithoutUTC := "2006-01-02 15:04:05"
-		timestamp2, err2 := time.Parse(formatDateWithoutUTC, timeString)
-		formatDateMilli := "2006-01-02 15:04:05 +0000 UTC" // hardcoded golang format date
-		timestamp3, err3 := time.Parse(formatDateMilli, timeString)
-		if err1 != nil && err2 != nil && err3 != nil {
-			iter.err = fmt.Errorf("could not parse timestamp: %v: %v, %v", nextVal["ts"], err1, err2)
+	value, err := iter.parseValue(nextVal["value"])
+	if err != nil {
+		iter.err = err
+		return false
+	}
+	ts, hasTimestamp := nextVal["ts"].(string)
+	var timestamp time.Time
+	if hasTimestamp {
+		timestamp, err = iter.parseTimestamp(ts)
+		if err != nil {
+			iter.err = err
 			return false
 		}
-		if err2 == nil {
-			timestamp = timestamp2
-		} else if err3 == nil {
-			timestamp = timestamp3
-		}
-		iter.cur = ResourceRecord{Entity: nextVal["entity"].(string), Value: nextVal["value"], TS: timestamp}
 	}
-
+	iter.cur = ResourceRecord{
+		Entity: nextVal["entity"].(string),
+		Value:  value,
+		TS:     timestamp,
+	}
 	return true
 }
 
+// Attempts to parse timestamp in one of the following formats:
+// 1. "2006-01-02 15:04:05.000000 UTC"
+// 2. "2006-01-02 15:04:05.000000"
+// 3. "2006-01-02 15:04:05.000000 +0000 UTC"
+// If any one of the three formats is valid, returns the parsed timestamp, otherwise it
+// returns an error
+func (iter *FileStoreFeatureIterator) parseTimestamp(ts string) (time.Time, error) {
+	// Attempt to parse timestamp as UTC first
```

Comment:
Could actually do this in a loop:

formats := []string{"", "", ""}
for format in formats {
  parsed, err := time.Parse(format, ts)
  if err == nil {
    return parsed, nil
  }
}
return time.Time{}, fmt.Errrorf(